### DISCUSSION TITLE
Allow Brick/Money instance on Currency Field

### DISCUSSION THREAD

**Discussion by nonovd** [laravel/nova-issues#3694] [Ideas & Feature Requests]
### Proposal

Add ability to directly pass an instance of `Brick/Money/Money` to _Currency Field_.

### Use Case

Custom Cast to `Brick/Money/Money`

```php
protected $casts = [
    'amount' => App\Casts\Money::class,
];
```

```php
Currency::make(__('Amount'), 'amount')
```

For now, we have to do:

```php
Currency::make(__('Amount'), 'amount')
    ->resolveUsing(fn ($value) => optional($value)->getAmount())
    ->displayUsing(fn ($value) => $value->formatTo(config('app.locale')))
```

**Comment by crynobone** [comment]
Looked at this today, while it was easy enough to set it up from model, filling the model using `Money` instance isn't. There's no way to verify that an attribute can accept `Money` or only int/float value
